### PR TITLE
Paid stats: sync wpcom site features

### PR DESCRIPTION
**Pull request by lsl**
Related to https://github.com/Automattic/red-team/issues/297

## Proposed changes:

- Moves Paid stats from Premium plans to Personal plans.
- Syncs with the changes in D167647-code (now merged).
- There should be no changes to Jetpack or even Jetpack stats with this change as this file is not leveraged by Jetpack stats (odyssey), this PR is just to keep the file in sync with wpcom.

### Other information:

P2: p1HpG7-v7f-p2

- [x] Have you written new tests for your changes, if applicable? See D167647-code
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

- Should be a no op, smoke test on a woa dev site with the wpcomsh plugin set to this branch using the Jetpack beta plugin
- See also: https://github.com/Automattic/wp-calypso/pull/97041 and D167647-code for testing plans around these changes (deployed).

### CODE REVIEW

File: projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php [hunk 1]
```diff
@@ -1149,22 +1150,25 @@ class WPCOM_Features {
...
+		self::STATS_PAID                        => array(
+			self::WPCOM_PERSONAL_AND_HIGHER_PLANS,
 			self::WP_P2_PLUS_MONTHLY,
-			self::WPCOM_PREMIUM_AND_HIGHER_PLANS,
 			self::JETPACK_STATS_PWYW,
 			self::JETPACK_STATS_MONTHLY,
 			self::JETPACK_STATS_BI_YEARLY,
 			self::JETPACK_STATS_YEARLY,
 			self::JETPACK_COMPLETE_PLANS,
+			self::JETPACK_BUSINESS_PLANS,
```
Comment (lsl) [inline projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php:1182]: @kangzj this was missing from the Jetpack version of this file, does it actually need adding here or does in fact need removing from the dotcom side?

Comment (kangzj) [reply projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php:1182]: I'm not entirely sure 🤔

I wonder if @Automattic/jetpack-martech folks could provide some insights.

Comment (IanRamosC) [reply projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php:1182]: Thanks for the ping, @kangzj. Based on the description in D158049-code, it seems like it actually needs adding here. Thanks for fixing that.

Comment (lsl) [reply projects/plugins/wpcomsh/wpcom-features/class-wpcom-features.php:1182]: It looks like odyssey stats sidesteps this class on atomic sites so it being missing doesn't matter, I'll sync this PR up to whatever we do in D167647-code for completeness but it should be a noop.

**Review by Initsogar — APPROVED** [review]
LGTM

**Pull request merged**